ui/app/AppLayouts/WalletV2/WalletHeader.qml

```diff
@@ -140,6 +229,479 @@ Item {
+                        }
+                    }
+
+                    GasSelector {
+                        id: gasSelector
+                        visible: !!selectedRecipientLoader.selectedRecipient || (!!recipientInput.text && recipientInput.valid)
+                        anchors.top: recipientInput.bottom
+                        anchors.topMargin: 16
+                        anchors.left: parent.left
+                        anchors.right: parent.right
+                        anchors.leftMargin: 36
+                        onVisibleChanged: {
+                            if (visible && assetAndAmount.isValid) {
+                                estimateGas()
+                            }
+                        }
+
+                        slowestGasPrice: parseFloat(walletModel.gasView.safeLowGasPrice)
+                        fastestGasPrice: parseFloat(walletModel.gasView.fastestGasPrice)
+                        getGasEthValue: walletModel.gasView.getGasEthValue
+                        getFiatValue: walletModel.balanceView.getFiatValue
+                        defaultCurrency: walletModel.balanceView.defaultCurrency
+                        property var estimateGas: Backpressure.debounce(gasSelector, 600, function() {
+                            if (!sendTransactionModal.recipient && assetAndAmount.selectedAsset && assetAndAmount.selectedAsset.address &&
+                                assetAndAmount.selectedAmount) return
+                            
+                            let gasEstimate = JSON.parse(walletModel.gasView.estimateGas(
+                                currentAccount.address,
+                                sendTransactionModal.recipient.address,
+                                assetAndAmount.selectedAsset.address,
+                                assetAndAmount.selectedAmount,
+                                ""))
+
+                            if (!gasEstimate.success) {
+                                console.warn(qsTrId("Error estimating gas: %1").arg(gasEstimate.error.message))
+                                return
+                            }
+                            selectedGasLimit = gasEstimate.result
+                        })
+                        gasSelectorButtonWidth: 125
+                        gasSelectorButtonHeight: 115
+                    }
+                }
+            }
+
+            GasValidatorV2 {
+                id: gasValidator
+                anchors.top: content.bottom
+                anchors.topMargin: 24
+                width: parent.width
+                visible: gasSelector.visible && !isValid
+                height: visible ? implicitHeight: 0
+                selectedAccount: sendTransactionModal.sender
+                selectedAmount: parseFloat(assetAndAmount.selectedAmount)
+                selectedAsset: assetAndAmount.selectedAsset
+                selectedGasEthValue: gasSelector.selectedGasEthValue
+            }
```

Comment:
Ideally, we'll end up with a proper `StatusValidator` in `GasSelector`, but to have that, we neeed to refactor `GasSelector` to use `StatusInput` first.

Also, `StatusInput` doesn't support asyncronous validators at the moment (which likely is needed as well)